docs/reference/query-builders.md

```diff
@@ -0,0 +1,382 @@
+
+If the value is `true` the query returns *only* docs in the archive. Note that permissions would still prevent a typical site visitor from obtaining any results, but an editor might.
+
+If the value is `null` (not `undefined`), the query returns docs regardless of archived status.
+
+### `areas()`
+
+```
+query.areas(false)
+
+query.areas([ 'thumbnail' ])
+```
+
+The `areas` builder controls whether to call the `load` methods of widget type managers for widgets in areas. `load` methods are used to do things such as populating relationships in widgets and other asynchronous actions when areas are loaded.
+
+The default value is `true`. Pass `false` to prevent the extra processing for a particular query. This is helpful to make queries faster when detailed area data is not needed.
+
+The value can also be an array containing strings representing area field names or dot paths to nested area fields. For example, with `.areas([ 'thumbnail' ])` you could load only the `'thumbnail'` area for all pages matching the query.
+
+### `attachments()`
+
+```
+query.attachments(true)
+```
+
+Passing `true` to the `attachments` builder will annotate all attachment fields in the returned documents with URLs and other metadata (primarily for images). This uses the the `apos.attachment.all` method with the `annotate: true` option. The builder defaults to `false`.
+
+### `autocomplete()`
+
+```
+query.autocomplete('tree house')
+```
+
+The `autocomplete` builder operates as a sort of fuzzy search for documents. It accepts a string and uses document search text properties to find matches. The string can contain multiple words, but each word (or segment) needs to be a partial string to find non-exact results. So "tree" will find results with "treehouse," but "treehouses" will not find "treehouse" (it is not meant as an advanced search tool, but can help find similar results).
+
+### `choices()`
+
+```
+query.choices('color, species')
+
+query.choices(['color', 'species'])
+```
+
+The `choices` builder is use to populate a `choices` object on the query results based on the filters passed to it. Filters can be passed in as a single string of comma-separated values or an array of strings.
+
+That `choices` object will contain valid options for each filter based on the rest of the query. Those options can be used to populate a select field for filtering an index page, for example.
+
+### `counts()`
+
+```
+query.choices('color, species')
+
+query.choices(['color', 'species'])
+```
+
+The `counts` builder works very similarly to the [`choices` builder](#choices). When passed a string of comma-separated filter names (or an array of filter names), it will include a `counts` object including the individual choices. Each choice will include a `count` property set to the number of documents in the query results that match that choice.
```

Comment:
After awaiting `query.toArray`, you can invoke `query.get('countsResults')` to get an object in which each property is the name of a field you passed to `choices` and each value is an array of choice objects with `label`, `value` and `count` properties.

Reply:
Again, server side you'd probably use `toChoices` with a single field name and an options object like `{ counts: true }`

Reply:
Similar situation to `choices`. This might be better not documented and we document `toChoices` in the upcoming guide.